test/e2e/testutils/util/get_test_variable.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2019-2021 The Liqo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	testutils "github.com/liqotech/liqo/pkg/utils"
+	cachedclientutils "github.com/liqotech/liqo/pkg/utils/cachedClient"
+)
+
+// GetEnvironmentVariable retrieves the value of the environment variable named by the key.
+// If the variable is not present calls klog.Fatal().
+func GetEnvironmentVariable(key string) string {
+	envVariable := os.Getenv(key)
+	if envVariable == "" {
+		klog.Fatalf("Environment variable '%s' not set", key)
+	}
+	return envVariable
+}
+
+// GetRestConfig retrieves the rest.Config from the kubeconfig variable.
+// If there is an error calls klog.Fatal().
+func GetRestConfig(kubeconfig string) *rest.Config {
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		klog.Fatal(err)
+	}
+	return config
+}
+
+// GetNativeClient creates a new Clientset for the given config.
+// If there is an error calls klog.Fatal().
+func GetNativeClient(config *rest.Config) *kubernetes.Clientset {
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		klog.Fatal(err)
+	}
+	return clientset
+}
+
+// GetControllerClient creates a new controller runtime client for the given config.
+// If there is an error calls klog.Fatal().
+func GetControllerClient(ctx context.Context, scheme *runtime.Scheme, config *rest.Config) client.Client {
+	controllerClient, err := cachedclientutils.GetCachedClientWithConfig(ctx, scheme, config)
```

Comment:
Is it necessary to use a cached client here? Using a standard one would remove the need for some Eventually